resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -249,6 +259,10 @@ private[spark] class Client(
     val capability = Records.newRecord(classOf[Resource])
     capability.setMemory(amMemory + amMemoryOverhead)
     capability.setVirtualCores(amCores)
+    if (amResources.nonEmpty) {
+      ResourceTypeHelper.setResourceInfoFromResourceTypes(amResources, capability)
+    }
+    logDebug("Created resource capability for AM request: %s".format(capability.toString))
```

Comment:
Nit: use interpolation, and `.toString` isn't needed

Reply:
fixed